lib/block_view_validator.go

```diff
@@ -0,0 +1,2133 @@
+		data = append(data, unstakedStaker.RawEncodeWithoutMetadata(blockHeight, skipMetadata...)...)
+	}
+
+	return data
+}
+
+func (txindexMetadata *UnregisterAsValidatorTxindexMetadata) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
+	var err error
+
+	// ValidatorPublicKeyBase58Check
+	validatorPublicKeyBase58CheckBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrapf(err, "UnregisterAsValidatorTxindexMetadata.Decode: Problem reading ValidatorPublicKeyBase58Check: ")
+	}
+	txindexMetadata.ValidatorPublicKeyBase58Check = string(validatorPublicKeyBase58CheckBytes)
+
+	// UnstakedStakers
+	numUnstakedStakers, err := ReadUvarint(rr)
+	if err != nil {
+		return errors.Wrapf(err, "UnregisterAsValidatorTxindexMetadata.Decode: Problem reading UnstakedStakers: ")
+	}
+	for ii := 0; ii < int(numUnstakedStakers); ii++ {
+		unstakedStaker := &UnstakedStakerTxindexMetadata{}
+		err = unstakedStaker.RawDecodeWithoutMetadata(blockHeight, rr)
+		if err != nil {
+			return errors.Wrapf(err, "UnregisterAsValidatorTxindexMetadata.Decode: Problem reading UnstakedStakers: ")
+		}
+		txindexMetadata.UnstakedStakers = append(txindexMetadata.UnstakedStakers, unstakedStaker)
+	}
+
+	return nil
+}
+
+func (txindexMetadata *UnregisterAsValidatorTxindexMetadata) GetVersionByte(blockHeight uint64) byte {
+	return 0
+}
+
+func (txindexMetadata *UnregisterAsValidatorTxindexMetadata) GetEncoderType() EncoderType {
+	return EncoderTypeUnregisterAsValidatorTxindexMetadata
+}
+
+//
+// TYPES: UnjailValidatorTxindexMetadata
+//
+
+type UnjailValidatorTxindexMetadata struct {
+}
+
+func (txindexMetadata *UnjailValidatorTxindexMetadata) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
+	return []byte{}
+}
+
+func (txindexMetadata *UnjailValidatorTxindexMetadata) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
+	return nil
+}
+
+func (txindexMetadata *UnjailValidatorTxindexMetadata) GetVersionByte(blockHeight uint64) byte {
+	return 0
+}
+
+func (txindexMetadata *UnjailValidatorTxindexMetadata) GetEncoderType() EncoderType {
+	return EncoderTypeUnjailValidatorTxindexMetadata
+}
+
+//
+// DB UTILS
+//
+
+func DBKeyForValidatorByPKID(validatorEntry *ValidatorEntry) []byte {
+	key := append([]byte{}, Prefixes.PrefixValidatorByPKID...)
+	key = append(key, validatorEntry.ValidatorPKID.ToBytes()...)
+	return key
+}
+
+func DBKeyForValidatorByStake(validatorEntry *ValidatorEntry) []byte {
+	key := append([]byte{}, Prefixes.PrefixValidatorByStake...)
+	key = append(key, EncodeUint8(uint8(validatorEntry.Status()))...)
+	// TotalStakeAmountNanos will never be nil here, but EncodeOptionalUint256
+	// is used because it provides a fixed-width encoding of uint256.Ints.
+	key = append(key, EncodeOptionalUint256(validatorEntry.TotalStakeAmountNanos)...)         // Highest stake first
+	key = append(key, EncodeUint64(math.MaxUint64-validatorEntry.RegisteredAtBlockHeight)...) // Oldest first
```

Comment:
Do we need this? I still haven't seen a good reason to include the RegisteredAtBlockHeight in the index key.

Reply:
Left it there. I think it's fine.